sklearn/impute/_knn.py

```diff
@@ -0,0 +1,270 @@
+import numpy as np
+
+from ..base import BaseEstimator, TransformerMixin
+from ..utils.validation import FLOAT_DTYPES
+from ..metrics import pairwise_distances
+from ..metrics.pairwise import _NAN_METRICS
+from ..neighbors.base import _get_weights
+from ..neighbors.base import _check_weights
+from ..utils import check_array
+from ..utils import is_scalar_nan
+from ..utils.mask import _get_mask
+from ..utils.validation import check_is_fitted
+
+
+class KNNImputer(TransformerMixin, BaseEstimator):
+    """Imputation for completing missing values using k-Nearest Neighbors.
+
+    Each sample's missing values are imputed using values from `n_neighbors`
+    nearest neighbors found in the training set. Each missing feature is then
+    imputed as the average, either weighted or unweighted, of these neighbors.
+    If a sample has more than one feature missing, then the neighbors for that
```

Comment:
I think this note is currently missing from the user guide. I would expect the guide to be more informative on such details than the description in the docstring.